torch/csrc/jit/script/compiler.cpp

```diff
@@ -2974,8 +2974,13 @@ std::unique_ptr<Function> CompilationUnit::define(
   auto creator = [def, _resolver, self](Function& method) {
     to_ir(def, _resolver, self, method);
   };
-  return torch::make_unique<Function>(
+  auto fn = torch::make_unique<Function>(
       name, is_optimized(), std::make_shared<Graph>(), creator);
+  if (self) {
+    // Register this as a method on `self`'s type
+    self->getClassType()->registerMethod(fn.get());
```

Comment:
If this is the _only_ place where getClassType is used, then I think it should be accomplished in a different way. compilation unit define is _not_ the right place for adding it as a method. It might be better to have the mega-define return a list of things it defined instead and let the actual class method compiler add it as a function.

Reply:
yeah that's fair—I was a little concerned that you could get them desynchronized if some random caller did `define` without knowing there was an associated `registerMethod` that was required, but this is prbably too intrusive

Reply:
Actually, I think we have to do it this way. Otherwise we won't have registered the methods when we called `ensure_defined()`, so methods won't be able to find each other.